packages/@aws-cdk/aws-kinesisfirehose/lib/delivery-stream.ts

```diff
@@ -211,12 +313,23 @@ export class DeliveryStream extends DeliveryStreamBase {
     });
     this.grantPrincipal = role;
 
+    if ((props.encryption === StreamEncryption.AWS_OWNED || props.encryption === StreamEncryption.UNENCRYPTED) && props.encryptionKey) {
+      throw new Error(`Specified stream encryption as ${StreamEncryption[props.encryption]} but provided a customer-managed key`);
+    }
+    const encryptionKey = props.encryptionKey ?? (props.encryption === StreamEncryption.CUSTOMER_MANAGED ? new kms.Key(this, 'Key') : undefined);
+    const encryptionConfig = (encryptionKey || (props.encryption === StreamEncryption.AWS_OWNED)) ? {
+      keyArn: encryptionKey?.keyArn,
+      keyType: encryptionKey ? 'CUSTOMER_MANAGED_CMK' : 'AWS_OWNED_CMK',
+    } : undefined;
+    encryptionKey?.grantEncryptDecrypt(role);
```

Comment:
need to add an `applyBefore(this)` so that we make sure the delivery stream does not start up before this grant has been applied

Reply:
I added this (also tried `Node.of(this).addDependency(keyGrant)` ) and when I synthesize a stack using encryption, then I get an error:
```
    throw new Error(`${instance} does not implement DependableTrait`);
```
I think it is something about the keyGrant causing this, not the DeliveryStream. Since other addDependencies work just fine. Still looking into it, @BenChaimberg let me know if you have ideas.

Reply:
Ben and I looked into this offline. There is something strange happening with key grants and it seems like there may be a bug in either kms or iam. I am working on getting a minimal repro of it, and will open up a separate issue for that. 

Right now, we don't need the DeliveryStream to depend on the key grant because data will not be lost. Firehose has a 24 hour period where it will keep and retry data if it gets access denied errors for the source encryption key. https://docs.aws.amazon.com/firehose/latest/dev/encryption.html

Reply:
Can you put a comment with that information in the code Madeline?

Reply:
Looks like @BenChaimberg added it